common/script/ops/revive.js

```diff
@@ -1,72 +1,107 @@
 import content from '../content/index';
 import i18n from '../i18n';
 import _ from 'lodash';
+import {
+  NotAuthorized,
+} from '../libs/errors';
+import splitWhitespace from '../libs/splitWhitespace';
 
-module.exports = function(user, req, cb, analytics) {
-  var analyticsData, base, cl, gearOwned, item, losableItems, lostItem, lostStat;
-  if (!(user.stats.hp <= 0)) {
-    return typeof cb === "function" ? cb({
-      code: 400,
-      message: "Cannot revive if not dead"
-    }) : void 0;
+module.exports = function revive (user, req = {}, analytics) {
+  if (user.stats.hp > 0) {
+    throw new NotAuthorized(i18n.t('cannotRevive', req.language));
   }
+
   _.merge(user.stats, {
     hp: 50,
     exp: 0,
-    gp: 0
+    gp: 0,
   });
+
   if (user.stats.lvl > 1) {
     user.stats.lvl--;
   }
-  lostStat = user.fns.randomVal(_.reduce(['str', 'con', 'per', 'int'], (function(m, k) {
+
+  let lostStat = user.fns.randomVal(_.reduce(['str', 'con', 'per', 'int'], function findRandomStat (m, k) {
```

Comment:
you can import randomVal from its file instead of using user.fns (we'll remove it), just remember to pass user as the first parameter